docs/code/GraphRunner.md

```diff
@@ -0,0 +1,123 @@
+﻿# JSON Graph format
+
+The entry point graph in TLC is an array of _nodes_. Each node is an object with the following fields:
+
+- _name_: string. Required. Name of the entry point.
+- _inputs_: object. Optional. Specifies non-default inputs to the entry point. 
+Note that if the entry point has required inputs (which is very common), the _inputs_ field is requred.
+- _outputs_: object. Optional. Specifies the variables that will hold the node's outputs.
+
+## Input and output types
+The following types are supported in JSON graphs:
+
+- _string_. Represented as a JSON string, maps to a C# string.
+- _float_. Represented as a JSON float, maps to a C# float or double.
+- _bool_. Represented as a JSON bool, maps to a C# bool.
+- _enum_. Represented as a JSON string, maps to a C# enum. The allowed values are those of the C# enum (they are also listed in the manifest).
+- _int_. Currently not implemented. Represented as a JSON integer, maps to a C# int or long.
+- _array_ of the above. Represented as a JSON array, maps to a C# array.
+- _dictionary_. Currently not implemented. Represented as a JSON object, maps to a C# `Dictionary<string,T>`.
+- _component_. Currently not implemented. Represented as a JSON object with 2 fields: _name_:string and _settings_:object.
+
+## Variables
+The following input/output types can not be represented as a JSON value:
+- _DataView_
```

Comment:
>_DataView_ [](start = 2, length = 10)

Is this usage intentional? There is no `DataView`, but there is an `IDataView`. Similar for file handles, the models, etc.

Reply:
Also if these are meant to be actual types, should they not be in `` ` `` backtick comments?

---
In reply to: [192868598](https://github.com/dotnet/machinelearning/pull/295#discussion_r192868598) [](ancestors = 192868598)